qemu/tests/virtio_serial_hotplug.py

```diff
@@ -0,0 +1,227 @@
+                    vm.virtio_ports.remove(_port)
+
+            if unplug_bus:
+                bus_id = params.get("bus_%s" % port)
+                error.context("start hot-deleting serial bus %s" % bus_id,
+                              logging.info)
+                device_del(bus_id, "type", "virtio-serial-pci")
+            chardevs.append(port_info["chardev"].strip('"'))
+            sub_type = params.get("sub_type_after_unplug")
+            if sub_type:
+                error.context(context_msg % (sub_type, "after hotunplug"),
+                              logging.info)
+                utils_test.run_virt_sub_test(test, params, env, sub_type)
+
+    def hotplug(hotplug_ports, driver, chardevs):
+        buses = filter_device_from_qtree("type", "virtio-serial-bus")
+        tested_port = []
+        for port in hotplug_ports:
+            chardev = chardevs.pop(-1)
+            bus = params.get("bus_%s" % port)
+            if not bus:
+                bus = buses[0]["id"]
+            else:
+                if bus not in buses:
+                    error.context("Start hot-adding serial bus %s" % bus,
+                                  logging.info)
+                    bus_driver = params.get("serial_bus_driver")
+                    device_add(bus_driver, bus)
+                    buses = filter_device_from_qtree("type", "virtio-serial-bus")
+                    for _bus in buses:
+                        if bus in _bus["id"]:
+                            bus = _bus["id"]
+            sub_type = params.get("sub_type_before_plug")
+            if sub_type:
+                error.context(context_msg % (sub_type, "before hotplug"),
+                              logging.info)
+                utils_test.run_virt_sub_test(test, params, env, sub_type)
+
+            error.context("Start hot-adding serial port %s" % port,
+                          logging.info)
+            name = port
+            device_id = port
+            device_add(driver, device_id, bus, chardev, name)
+            tested_port.append(port)
+            filename = filter_chardev_filename(vm, chardev)
+            if params.get('virtio_port_type') in ("console",
+                                                  "virtio_console"):
+                vm.virtio_ports.append(
+                    qemu_virtio_port.VirtioConsole(name, name,
+                                                   filename))
+            else:
+                vm.virtio_ports.append(
+                    qemu_virtio_port.VirtioSerial(name, name,
+                                                  filename))
+
+            sub_type = params.get("sub_type_after_plug")
+            if sub_type:
+                error.context(context_msg % (sub_type, "after hotplug"),
+                              logging.info)
+                params["file_transfer_serial_ports"] = " ".join(tested_port)
+                utils_test.run_virt_sub_test(test, params, env, sub_type)
```

Comment:
Hi @FengYang 

For the func run_virt_sub_test(), the definition is:
def run_virt_sub_test(test, params, env, sub_type=None, tag=None):
    :param sub_type: Type of called test script.

When we run subtest "virtio_serial_file_transfer", it would find a script instead the func().

Error Info:
13:28:52 INFO | Context: Running sub test 'virtio_serial_file_transfer' after hotplug
13:28:52 ERROR| Test failed: TestError: Could not find test file virtio_serial_file_transfer.py on directories